nestkernel/sparse_node_array.h

```diff
@@ -39,52 +39,100 @@ namespace nest
 class Node;
 
 /**
- * Provide sparse representation of local nodes.
+ * Sparse representation of local nodes.
  *
- * This class is a container providing lookup of local nodes (as Node*)
- * based on node IDs.
+ * SparseNodeArray maps node IDs of thread-local nodes to Node*. It provides
+ * a const iterator interface for iteration over all local nodes, lookup by
+ * node ID and lookup by numeric index into local nodes. The latter is provided
+ * to support HPC synapses using TargetIdentifierIndex representation.
  *
- * Basically, this array is a vector containing only pointers to local nodes.
- * For M MPI processes, we have
+ * For efficient lookup, all normal nodes (with proxies) and all devices need
```

Comment:
Does normal nodes mean nodes that represent neurons?

```suggestion
 * For efficient lookup, all normal nodes (with proxies) need to be created together 
```

Reply:
Yes, "normal" means (with minimal exceptions) nodes representing neurons.